src/libratbag.h

```diff
@@ -430,6 +430,15 @@ ratbag_device_get_user_data(const struct ratbag_device *device);
 const char *
 ratbag_device_get_name(const struct ratbag_device* device);
 
+/**
+ * @ingroup device
+ *
+ * @param device A previously initialized ratbag device
+ * @return "mouse" or "keyboard"
+ */
+const char *
+ratbag_device_get_device_type(const struct ratbag_device* device);
```

Comment:
as already said above, much easier to make the enum a public one and return a limited set here. This also helps with things like internationalization, `TYPE_MOUSE` is always that, but the string "mouse" is only that in english.